packages/core/plugins/core-plugins-contracts-server/README.md

```diff
@@ -0,0 +1,5 @@
+# @kbn/core-plugins-contracts-server
+
+This package contains the public types for core's server-side plugins service's contracts.
+
+The dedicated package was required to avoid a cyclic dependencies between the `plugins` and `lifecycle` domains.
```

Comment:
As explained, I faced a cyclic dependency issue between these two domains (the plugin context depends on lifecycle for coreSetup/coreStart, so lifecycle can't depend on the plugin types package), so I had to create dedicated packages for the contract types

Reply:
Can't we just use cycle dependencies here? :trollface: